### PR TITLE
job submit 01 removes old 02+ logs

### LINKED ISSUE #480: Remote run-directory housekeeping.
**For task polling to work under all circumstances, old task-status files have to be cleaned up at suite start-up**. Since #453 we can do this housekeeping on the suite host, but it is currently turned off by default for Rose compatibility, and it has not been implemented for task hosts.

The following link details a specific case where task polling can get the wrong result if old task status files are left lying about from a previous suite run: https://github.com/cylc/cylc/pull/462#issuecomment-18185769.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Should be enough to close #480.

### CODE REVIEW

**Comment by hjoliver** [comment]
@aosprey - this closes an old issue you raised https://github.com/cylc/cylc/pull/462#issuecomment-18155564 (sorry for the delay!)

File: tests/job-submission/12-tidy-submits-of-prev-run.t [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test job submission rundb task_jobs entry with garbage host command.
```
Comment (hjoliver) [inline tests/job-submission/12-tidy-submits-of-prev-run.t]: Comment refers to another test.

**Comment by hjoliver** [comment]
Review 1 (possibly premature) - good, works, one comment to fix.

File: tests/job-submission/12-tidy-submits-of-prev-run.t (hunk 1 above)
Comment (matthewrmshin) [inline tests/job-submission/12-tidy-submits-of-prev-run.t]: Comment fixed.

**Comment by matthewrmshin** [comment]
I have added more tests for remote host and remote host with shared file system.

By adding the test for remote host (without shared file system), I have discovered that this change is insufficient as the old job log directories on the suite host will still be left untouched. The best place to delete these, I believe, is around the time when the job file is written. Unfortunately, changing this part of the code, however, is likely to create a very difficult conflict with my other PR #1880, so I'll deal with this remaining issue as part of #1880 once this PR is merged.

**Comment by matthewrmshin** [comment]
@benfitzpatrick please review 2.

**Comment by hjoliver** [comment]
> the old job log directories on the suite host will still be left untouched.

Ah, yes (I don't currently have access to my remote hosts...)

**Comment by hjoliver** [comment]
Old remote-task status files left on the suite host will not trigger the polling error of #480, so this is still sufficient to close the issue.

**Comment by hjoliver** [comment]
The new tests look fine. Over to @benfitzpatrick.

**Comment by benfitzpatrick** [comment]
Looks fine.

**Pull request merged**